test/cypress/integration/simplified-account/service-settings/card-payments/card-payments.cy.js

```diff
@@ -0,0 +1,157 @@
+          defaultBillingAddressCountry: 'GB',
+          allowApplePay: true,
+          allowGooglePay: true,
+          serviceName: 'version 1'
+        })
+        cy.visit(pageUrl)
+        assertSummaryListsEqual([
+          [
+            {
+              key: 'Collect billing address',
+              value: 'On'
+            },
+            {
+              key: 'Default billing address country',
+              value: 'United Kingdom'
+            }
+          ],
+          [
+
+            {
+              key: 'Apple Pay',
+              value: 'On'
+            },
+            {
+              key: 'Google Pay',
+              value: 'On'
+            }
+          ]
+        ])
+      })
+
+      it('should display the provided card payment details (version 2 -everything off)', () => {
+        setupStubs({
+          collectBillingAddress: false,
+          defaultBillingAddressCountry: 'IE',
+          allowApplePay: false,
+          allowGooglePay: false,
+          serviceName: 'version 2'
+        })
+
+        cy.visit(pageUrl)
+        assertSummaryListsEqual([
+          [
+            {
+              key: 'Collect billing address',
+              value: 'Off'
+            },
+            {
+              key: 'Default billing address country',
+              value: 'IE'
+            }
+          ],
+          [
+
+            {
+              key: 'Apple Pay',
+              value: 'Off'
+            },
+            {
+              key: 'Google Pay',
+              value: 'Off'
+            }
+          ]
+        ])
+      })
+    })
+
+    // describe('for a non-admin user', () => {
+    //   beforeEach(() => {
+    //     setupStubs({
+    //       role: 'view-and-refund'
+    //     })
+    //   })
+    //
+    //   it('should return a 403', () => {
+    //     cy.request({
+    //       url: pageUrl,
+    //       failOnStatusCode: false
+    //     }).then(response => expect(response.status).to.eq(403))
+    //   })
+    // })
+  })
+})
+
+function assertSummaryListsEqual (expectation) {
+  expectation.forEach((currentSummaryListExpectations, summaryListIndex) => {
+    currentSummaryListExpectations.forEach((currentRowExpectations, currentRowIndex) => {
+      const currentRowExpectationsKeys = Object.keys(currentRowExpectations)
+      if (!currentRowExpectationsKeys.includes('key') || !currentRowExpectationsKeys.includes('value') || currentRowExpectationsKeys.lenght > 2) {
+        throw new Error(`Row expectations *must* (currently) be "key" and "value".  [${currentRowExpectationsKeys.join(', ')}] provided`)
+      }
+      cy.get('.govuk-summary-list').eq(summaryListIndex).within(() => {
+        cy.get('.govuk-summary-list__row').eq(currentRowIndex).within(() => {
+          cy.get('.govuk-summary-list__key').should('contain', currentRowExpectations.key)
+          cy.get('.govuk-summary-list__value').should('contain', currentRowExpectations.value)
+        })
+      })
+    })
+  })
+}
```

Comment:
I think this is overly DRY and inlining the assertions would be clearer

Reply:
The reason I did this was for readability within the test itself:

```
        assertSummaryListsEqual([
          [
            {
              key: 'Collect billing address',
              value: 'On'
            },
            {
              key: 'Default billing address country',
              value: 'United Kingdom'
            }
          ],
          [

            {
              key: 'Apple Pay',
              value: 'On'
            },
            {
              key: 'Google Pay',
              value: 'On'
            }
          ]
        ])
      })
```

What I was trying to avoid was two blocks that look like this as the important detail is hidden in the numbers inside `eq()`:

```
        cy.get('.govuk-summary-list').within(() => {
          cy.get('.govuk-summary-list__key').eq(0).should('contain', 'Ask users for their email address')
          cy.get('.govuk-summary-list__value').eq(0).should('contain', 'On')
          cy.get('.govuk-summary-list__actions a').eq(0).should('not.contain', 'Change')
          cy.get('.govuk-summary-list__key').eq(1).should('contain', 'Send payment confirmation emails')
          cy.get('.govuk-summary-list__value').eq(1).should('contain', 'On')
          cy.get('.govuk-summary-list__actions a').eq(1).should('not.contain', 'Change')
          cy.get('.govuk-summary-list__key').eq(2).should('contain', 'Send refund emails')
          cy.get('.govuk-summary-list__value').eq(2).should('contain', 'On')
          cy.get('.govuk-summary-list__actions a').eq(2).should('not.contain', 'Change')
        })
```

Reply:
I'm not sure I understand what you mean, changing the numbers in `eq()` from being explicitly stated to being an array index doesn't (to my mind) elucidate any detail that would otherwise be hidden. In the example block above a few line breaks could improve the readability, along with perhaps the use of e.g. `.eq(0).within(() => {})`, which admittedly is less dry but in the event of a regression would make it very easy to identify where the regression occured with only a line number for the failed assertion.